scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -398,91 +397,103 @@ def info(what):
-                    # load platform directly, otherwise, iterate the directory to
-                    # get all valid board revision based on each <board>_<revision>.conf.
-                    if '@' not in platform.name:
-                        tmp_dir = os.listdir(os.path.dirname(file))
-                        for item in tmp_dir:
-                            # Need to make sure the revision matches
-                            # the permitted patterns as described in
-                            # cmake/modules/extensions.cmake.
-                            revision_patterns = ["[A-Z]",
-                                                    "[0-9]+",
-                                                    "(0|[1-9][0-9]*)(_[0-9]+){0,2}"]
-
-                            for pattern in revision_patterns:
-                                result = re.match(f"{platform.name}_(?P<revision>{pattern})\\.conf", item)
-                                if result:
-                                    revision = result.group("revision")
-                                    yaml_file = f"{platform.name}_{revision}.yaml"
-                                    if yaml_file not in tmp_dir:
-                                        platform_revision = copy.deepcopy(platform)
-                                        revision = revision.replace("_", ".")
-                                        platform_revision.name = f"{platform.name}@{revision}"
-                                        platform_revision.normalized_name = platform_revision.name.replace("/", "_")
-                                        platform_revision.default = False
-                                        self.platforms.append(platform_revision)
-
-                                    break
-
-
-                except RuntimeError as e:
-                    logger.error("E: %s: can't load: %s" % (file, e))
+                    try:
+                        scp = TwisterConfigParser(file, Platform.platform_schema)
+                        sdata = scp.load()
+                        datas.append(sdata)
+                    except Exception as e:
+                        logger.error(f"Error loading {file}: {e!r}")
+                        self.load_errors += 1
+                        continue
+                bdirs[board.dir] = datas
+            data = {}
+            if os.path.exists(board.dir / 'twister.yaml'):
+                try:
+                    scp = TwisterConfigParser(board.dir / 'twister.yaml', Platform.platform_schema)
+                    data = scp.load()
+                except Exception as e:
+                    logger.error(f"Error loading {board.dir / 'twister.yaml'}: {e!r}")
                     self.load_errors += 1
+                    continue
+                new_config_found = True
+
+            # helper function to initialize and add platforms
+            def init_and_add_platforms(data, board, target, qualifier, aliases):
```

Comment:
this definition comes in the middle of the `for` loop. please, move it elsewhere